### PR TITLE
[Seneye] Bug fix for using Pond or Home sensors.

### PR DESCRIPTION
**Pull request by MickUrsell**
This is a bug fix for Seneye binding when using Pond or Home Sensors.
Previously the binding would fail as it would try and pull readings for
Channels that didn't exist on these sensors.

Signed-off-by: MickUrsell  <bindings@mru-photography.co.uk>

### CODE REVIEW

File: bundles/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/internal/handler/SeneyeHandler.java [hunk 1]
```diff
@@ -69,21 +69,50 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
-            updateState(CHANNEL_LUX, new DecimalType(readings.lux.curr));
-            updateState(CHANNEL_KELVIN, new DecimalType(readings.kelvin.curr));
-            updateState(CHANNEL_LASTREADING, new DateTimeType(readings.status.getLast_experimentDate()));
-            updateState(CHANNEL_SLIDEEXPIRES, new DateTimeType(readings.status.getSlide_expiresDate()));
-            updateState(CHANNEL_WRONGSLIDE, new StringType(readings.status.getWrong_slideString()));
-            updateState(CHANNEL_SLIDESERIAL, new StringType(readings.status.getSlide_serialString()));
-            updateState(CHANNEL_OUTOFWATER, new StringType(readings.status.getOut_of_waterString()));
-            updateState(CHANNEL_DISCONNECTED, new StringType(readings.status.getDisconnectedString()));
+            if (seneyeService.seneyeType == 3) {
+                logger.debug("found that sensor is type 3");
```
Comment (nikotanghe) [inline bundles/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/internal/handler/SeneyeHandler.java:73]: Maybe log what type 1,2 or 3 actually is ? Pond/Home/Reef

Comment (MickUrsell) [reply bundles/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/internal/handler/SeneyeHandler.java]: Hi Niko,

Many thanks for the feedback. I have changed the log as suggested.
This binding has been great for me, thanks for creating it.

File: bundles/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/internal/handler/SeneyeHandler.java [hunk 2]
```diff
@@ -69,21 +69,50 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                updateState(CHANNEL_TEMPERATURE, new DecimalType(readings.temperature.curr));
+                updateState(CHANNEL_NH3, new DecimalType(readings.nh3.curr));
+                updateState(CHANNEL_PH, new DecimalType(readings.ph.curr));
+                updateState(CHANNEL_LASTREADING, new DateTimeType(readings.status.getLast_experimentDate()));
+                updateState(CHANNEL_SLIDEEXPIRES, new DateTimeType(readings.status.getSlide_expiresDate()));
+                updateState(CHANNEL_WRONGSLIDE, new StringType(readings.status.getWrong_slideString()));
+                updateState(CHANNEL_SLIDESERIAL, new StringType(readings.status.getSlide_serialString()));
+                updateState(CHANNEL_OUTOFWATER, new StringType(readings.status.getOut_of_waterString()));
+                updateState(CHANNEL_DISCONNECTED, new StringType(readings.status.getDisconnectedString()));
+            }
```
Comment (cpmeister) [inline bundles/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/internal/handler/SeneyeHandler.java:113]:
```suggestion
            logger.debug("Updating readings for sensor type {}", seneyeService.seneyeType);
            switch (seneyeService.seneyeType) {
                case 3: 
                    updateState(CHANNEL_NH4, new DecimalType(readings.nh4.curr));
                    updateState(CHANNEL_PAR, new DecimalType(readings.par.curr));
                    updateState(CHANNEL_LUX, new DecimalType(readings.lux.curr));
                    updateState(CHANNEL_KELVIN, new DecimalType(readings.kelvin.curr));
                case 2:
                    updateState(CHANNEL_O2, new DecimalType(readings.o2.curr));
                case 1:
                    updateState(CHANNEL_TEMPERATURE, new DecimalType(readings.temperature.curr));
                    updateState(CHANNEL_NH3, new DecimalType(readings.nh3.curr));
                    updateState(CHANNEL_PH, new DecimalType(readings.ph.curr));
                    updateState(CHANNEL_LASTREADING, new DateTimeType(readings.status.getLast_experimentDate()));
                    updateState(CHANNEL_SLIDEEXPIRES, new DateTimeType(readings.status.getSlide_expiresDate()));
                    updateState(CHANNEL_WRONGSLIDE, new StringType(readings.status.getWrong_slideString()));
                    updateState(CHANNEL_SLIDESERIAL, new StringType(readings.status.getSlide_serialString()));
                    updateState(CHANNEL_OUTOFWATER, new StringType(readings.status.getOut_of_waterString()));
                    updateState(CHANNEL_DISCONNECTED, new StringType(readings.status.getDisconnectedString()));
            }
```

**Comment by cpmeister** [comment]
Manually checked sign-off

**Pull request merged**